src/FakeItEasy/Core/StrictFakeRule.cs

```diff
@@ -1,14 +1,43 @@
 namespace FakeItEasy.Core
 {
+    using static ObjectMembers;
+
     internal class StrictFakeRule : IFakeObjectCallRule
     {
+        private readonly StrictFakeOptions options;
+
+        public StrictFakeRule(StrictFakeOptions options)
+        {
+            this.options = options;
+        }
+
         public int? NumberOfTimesToCall => null;
 
-        public bool IsApplicableTo(IFakeObjectCall fakeObjectCall) => true;
+        public bool IsApplicableTo(IFakeObjectCall fakeObjectCall)
+        {
+            if (fakeObjectCall.Method.IsSameMethodAs(EqualsMethod))
+            {
+                return !this.HasOption(StrictFakeOptions.AllowEquals);
+            }
+
+            if (fakeObjectCall.Method.IsSameMethodAs(GetHashCodeMethod))
+            {
+                return !this.HasOption(StrictFakeOptions.AllowGetHashCode);
+            }
+
+            if (fakeObjectCall.Method.IsSameMethodAs(ToStringMethod))
+            {
+                return !this.HasOption(StrictFakeOptions.AllowToString);
+            }
+
+            return true;
+        }
 
         public void Apply(IInterceptedFakeObjectCall fakeObjectCall)
         {
             throw new ExpectationException(ExceptionMessages.CallToUnconfiguredMethodOfStrictFake(fakeObjectCall));
         }
+
+        private bool HasOption(StrictFakeOptions flag) => (flag & this.options) == flag;

```

Comment:
Because `Enum.HasFlag` is slow (it works on `Enum` rather than a specific enum, which causes the argument value to be boxed)

Reply:
I did not know this! Thanks.